content/nats-certificate-rotation.md

```diff
@@ -0,0 +1,216 @@
+# Rotating NATS Certificates
+
+The following strategy rotates the certificates and CA across the director, health monitor, NATS, and all VMs. See [Components of Bosh](bosh-components.md) for more information on core components.
+
+
+![image](images/nats_rotation.png)
+
+#### Preconditions:
+
+* Director is in a healthy state.
+* All VMs are in `running` state in all deployments.
+
+
+### Redeploy the director, health monitor, and NATS, with a new CA.
+
+```shell
+
+$ bosh create-env ~/workspace/bosh-deployment/bosh.yml \
+ --state ./state.json \
+ -o ~/workspace/bosh-deployment/[IAAS]/cpi.yml \
+ -o add-new-ca.yml \
+ -o ... additional opsfiles \
+ --vars-store ./creds.yml \
+ -v ... additional vars
+```
+
+* Adds new variables for the new CA/certificates/private_key. 
+* The director and health monitor are given a modified CA with the original CA and the new CA concatenated as `((nats_server_tls.ca))((nats_server_tls_2.ca))`. This allows the director to communicate with NATS using new certificates, while NATS can still communicate with the director.[1](#limitations).
+* The director and health monitor are updated to use certificates and keys generated by the new CA.
+* NATS continues to use the old certificates and private key. NATS is given the concatenated CA from above to verify certificate generated using new CA and old CA. 
+* Each VM/agent continues to use the old certificates to communicate with NATS. 
+* 
+
+`add-new-ca.yml` 
+
+```yaml
+---
+- type: replace
+  path: /instance_groups/name=bosh/properties/nats/tls/ca?
+  value: ((nats_server_tls.ca))((nats_server_tls_2.ca))
```

Comment:
This probably, technically, works as is. But we should probably explicitly include a newline between the values since a newline is not required at the end of a certificate (and external tooling or credhub may be doing it differently). I think?